packages/providers/src/provider.ts

```diff
@@ -580,7 +676,7 @@ export default class Provider {
       filter: { owner: owner.toB256(), assetId: assetId && hexlify(assetId) },
     });
 
-    const coins = result.coins.edges!.map((edge) => edge!.node!);
+    const coins = result.coins.edges.map((edge) => edge.node);
```

Comment:
Are these changes unrelated?

Reply:
I was getting warnings that these no-null-assertions are unnecessary so I removed them.